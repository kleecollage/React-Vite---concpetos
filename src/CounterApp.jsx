import { useState } from 'react'
import { PropTypes } from "prop-types"

const handleAdd = (event, newValue) => {
  console.log(event, newValue);
}

export const CounterApp = ({ value }) => {
    
    const [counter, setCounter] = useState(value);
    const counterAdd = () => {
        // setCounter(counter + 1);
        setCounter((c) => c + 1);
        console.log('+1');
    }
    const counterMinus = () => setCounter(counter - 1);
    const counterReset = () => setCounter(value)

  return (
    <>
      <h1>CounterApp: </h1>
      <h2 id="counter">{counter}</h2>
          <button onClick={ (event) => handleAdd(event, 'Hola') }>Button</button>
          <button onClick={ counterAdd }>
              +1
          </button>
          <button onClick={ counterMinus }>
              - 1
          </button>
          <button aria-label='btn-reset' onClick={ counterReset }>
              Reset
          </button>
    </>
  );
}

CounterApp.propTypes = {
    value: PropTypes.number
}
// defaultProps dejara de usarse en siguientes versiones de React
// los default props deben ser definidos en el main
// tambien pueden ser definidos en la funcion, directamente como una desestructuracion
// CounterApp.defaultProps = {
//     value:7
// }