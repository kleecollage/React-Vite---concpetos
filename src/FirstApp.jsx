import { PropTypes } from "prop-types"; // se instala con yarn add prop-types o si se usa npm npm intall prop-types

const newMessage = 'Soy la constante newMessage';
const num = 455;
const num2 = 333;
const bool = true;
const arr = [1, 2, 3, 4, 5, 6, 7, 8, 9];
const obj = {
    message: 'Hola Mundo',
    title: 'Soy un objeto'
}

const suma = () => {
    let resultado = num + num2;
    return resultado
}

// export default function FirstApp(props) { // Lo comun es usar desestructuracion....
export default function FirstApp({
    title = 'title: Soy un default value property',
    subTitle = 'subtitle: soy un default prop',
    propValue = 5,
    name}) {
    // if (!propValue) {
    //     throw new Error('porpValue no existe')
    // }
    // console.log(props) // props = properties
    return (
        // el atajo <> es l aabreviatura de fragment y nos evita el tener que importarlo
        <> 
            {/* <h1>{title}</h1>
            <h2>{subTitle + 1}</h2>
            <h1>FirstApp</h1>
            <h2>by Antonio Hernandez Santander</h2>
            <h3>{3 * 2}</h3>
            <h4>{newMessage}</h4>
            <h5>{num}</h5>
            <h6>{bool}</h6>
            <p>{arr}</p>
            <code>{JSON.stringify (obj)}</code>
            <div>{ suma() }</div> */}
            <h1 data-testid='test-title'>{title}</h1>
            <div>{subTitle}</div>
            <div>{subTitle}</div>
            <h3>{name}</h3>
            <p>{propValue}</p>
        </>
  )
}

FirstApp.propTypes = {
    propValue: PropTypes.number.isRequired,
    subTitle: PropTypes.string.isRequired 
}

// FirstApp.defaultProps = {
//   subTitle: "subTitle (default): valor definido en defaultProps",
//   propValue: 0,
//   name: "name (default): Antonio Hernandez",
// };

