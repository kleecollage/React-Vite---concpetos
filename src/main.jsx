import React from 'react';
import ReactDOM from 'react-dom/client';
import { HelloWorldApp } from './HelloWorldApp';
import './styles.css';
import {CounterApp} from './CounterApp';
import FirstApp from './FirstApp';

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <HelloWorldApp />
    <CounterApp value={73}/>
    {/* <FirstApp title='title default desde el main'/> */}
  </React.StrictMode>
);