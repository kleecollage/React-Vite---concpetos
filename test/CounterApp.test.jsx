import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe('Prueba de CounterApp', () => {
  // *******************************************************
  test("debe de hacer match con el snapshot", () => {
    const { container } = render(<CounterApp />);
    expect(container).toMatchSnapshot();
  });
  // *******************************************************
  test("debe de mostrar el valor inicial de 73", () => {
    render(<CounterApp value={73} />);
    expect(screen.getByText(73)).toBeTruthy();
  });
  // *******************************************************
  test("debe incrementar con el boton +1", () => {
    render(<CounterApp value={73} />);
    fireEvent.click(screen.getByText("+1"));
    expect(screen.getByText(74)).toBeTruthy();
  });
  // *******************************************************
  test("debe decrecer con el boton -1", () => {
    render(<CounterApp value={73} />);
    fireEvent.click(screen.getByRole("button", {name: '- 1'}));
    expect(screen.getByText("72")).toBeTruthy();
  });
  // *******************************************************
  test("el boton reset debe funcionar", () => {
    render(<CounterApp value={73} />);
    fireEvent.click(screen.getByRole("button", { name: "- 1" }));
    fireEvent.click(screen.getByRole("button", { name: "- 1" }));
    fireEvent.click(screen.getByRole("button", { name: "- 1" }));
    fireEvent.click(screen.getByRole("button", { name: "btn-reset" }));
    expect(screen.getByText("73")).toBeTruthy();

  });
})