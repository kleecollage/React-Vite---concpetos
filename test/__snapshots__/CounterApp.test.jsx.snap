// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de CounterApp debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp: 
  </h1>
  <h2
    id="counter"
  />
  <button>
    Button
  </button>
  <button>
    +1
  </button>
  <button>
    - 1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
